src/rust/engine/options/src/build_root.rs

```diff
@@ -59,6 +60,16 @@ impl BuildRoot {
             ))?;
         }
     }
+
+    pub fn convert_to_string(&self) -> Result<String, String> {
```

Comment:
Named `convert_to_string` and not `to_string` since the latter by convention (and as required by the ToString trait) returns a String, not a Result.

Reply:
Also, we still only call this in one place, so it wasn't strictly necessary to move it, but I had moved it when possibly needing to call it from multiple places, and this felt like a better place for it.